### DISCUSSION TITLE
Add Custom Tag To SQL Auto-Instrumentation

### DISCUSSION THREAD

**Discussion by bsulliva** [open-telemetry/opentelemetry-dotnet#4769] [Q&A] [answered]
Hello, I want to add custom tags to individual SQL queries that are auto-instrumented by the OpenTelemetry.Instrumentation.SqlClient library. I tried adding this, Activity.Current?.SetTag("sql.tagwith", "CreateCartAsync");, right before the EF Core code executes, but it doesn't show up. How can I add a custom tag to auto-instrumented SQL traces?

**Comment by vishweshbankwar** [comment, marked as answer]
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.SqlClient#enrich

**Comment by bsulliva** [comment]
Thanks @vishweshbankwar. This appears to be enriching at a global scope, unless I'm missing something? I was looking for something to enrich at an individual SQL call.

**Reply by vishweshbankwar** [reply]
Enrich is called for each outgoing SQL call.

**Comment by bsulliva** [comment]
Ok thanks. Does the library support enriching/adding custom tags for individual SQL calls?
